### PR TITLE
Haar 2498 proxy calls to nomis-users-roles-api

### PR DESCRIPTION
**Pull request by thomasridd**
### What
Allow hmpps-manage-users to ditch its nomis-user-roles-api dependency and route through this api 

### How
Add the new nomis endpoints under /prisonusers/, - adding tests, and integrations

This PR shouldn't be too controversial in that there is nearly no overlap with existing code. The exception to this is ktlint is getting tetchy and requiring some formatting changes in code I had otherwise not touched. Where I have made apparently pointless formatting changes (particularly `UserController.kt`) the reason is they were failing the lint check

### CODE REVIEW

File: docker-compose-dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+version: '3.1'
```
Comment (ravmoj) [inline docker-compose-dependencies.yml:1]: this service  already have one  docker file 'docker-compose-full.yml'

Comment (thomasridd) [reply docker-compose-dependencies.yml:1]: docker-compose-full loads up manage-users-api itself. docker-compose-dependencies is there for dev purposes and doesn't load the service itself. I'll add a comment to it.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/model/filter/PrisonUserFilter.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.justice.digital.hmpps.nomisuserrolesapi.data.filter
```
Comment (ravmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/model/filter/PrisonUserFilter.kt:1]: package doesn't match

Comment (thomasridd) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/model/filter/PrisonUserFilter.kt:1]: very good point - I'm surprised nothing automatic picked this up

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/model/PrisonAccountStatus.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  val desc: String,
+  val isExpired: Boolean,
+  val isLocked: Boolean,
+  val isGracePeriod: Boolean,
+  /**
+   * whether the user has locked themselves out by getting password incorrect in c-nomis
+   */
+  val isUserLocked: Boolean,
+) {
+  OPEN(0, "OPEN", false, false, false, false),
```
[the remark this answers is not in the document]
Comment (thomasridd) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/model/PrisonAccountStatus.kt:13]: I'll pull out the companion object but I think the enum content has value in that it helps document the api

[the remark this answers is not in the document]
Comment (thomasridd) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/model/PrisonAccountStatus.kt:10]: cancel the pulling out of the companion object - without it there are a lot of deprecation notices during linting about getters for enums being deprecated - this code fixes that

File: docker-compose-dependencies.yml (hunk 1 above)
Comment (thomasridd) [reply docker-compose-dependencies.yml]: I've removed it - I can run docker-compose-full with specific services

**Review by ravmoj — APPROVED** [review]
lgtm

**Pull request merged**